src/services/identity/NotificationsService.ts

```diff
@@ -218,31 +196,30 @@ class NotificationsService {
           model: Site,
           as: "site",
           required: true,
-          include: [
-            {
-              model: Repo,
-              required: true,
-              where: {
-                name: siteName,
-              },
-            },
-          ],
+          where: {
+            id: siteMember.siteId,
+          },
         },
       ],
     })
 
     if (recentTargetNotification) {
       // Update existing notification
-      await recentTargetNotification.update({
-        firstReadTime: null,
-        createdAt: new Date(),
-      })
+      // createdAt is a special column which must be flagged as changed
+      recentTargetNotification.changed("createdAt", true)
+      await recentTargetNotification.update(
+        {
+          firstReadTime: null,
+          createdAt: new Date(),
```

Comment:
could i get some clarity on this? i'm not entirely sure why we are erasing these properties and setting `createdAt` to now

Reply:
We're updating an existing notification so that we don't flood the user with notifications if many of the same type are created within a short period of time - the desired behaviour is for the past notification to be updated instead, and we simulate this by updating the `createdAt` time, which is used to indicate the time of notification

Reply:
actually i'm still not very sure of this HAHA

it seems that we're finding a recent notification targeted at the current user and setting it to be unread? but wouldn't similar notifs (but with different messages) then be hidden?

Reply:
We're also looking for a notif of the same type - currently those all share the same message, but with the only difference being the username in the message, which should be okay to overwrite, so losing the content of the original message shouldn't be a problem